<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="hand4">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="-0.0936672124273331 1.63489999316369E-10 0.535231939309104"
        rpy="0 0 0" />
      <mass
        value="279.343309739745" />
      <inertia
        ixx="34.9281247022782"
        ixy="-8.13293010176962E-09"
        ixz="-2.07006228380223"
        iyy="35.7141019454583"
        iyz="-2.13551763088024E-09"
        izz="12.2415875307062" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.067 0.106 0.643 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="finger_1">
    <inertial>
      <origin
        xyz="0.0582113837489129 4.16223523425074E-08 4.90502332084475E-08"
        rpy="0 0 0" />
      <mass
        value="2.62291219261925" />
      <inertia
        ixx="0.00599150094431563"
        ixy="2.04568885529762E-08"
        ixz="1.56227476558711E-08"
        iyy="0.0153590426966864"
        iyz="-1.69856752757797E-07"
        izz="0.0188065552886401" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.067 0.106 0.643 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin1"
    type="revolute">
    <origin
      xyz="0.330644040002145 0.175 1.18915890545841"
      rpy="3.14159265358979 -0.582486579623969 0" />
    <parent
      link="base_link" />
    <child
      link="finger_1" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1"
      upper="1"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_1_1">
    <inertial>
      <origin
        xyz="0.109230114297313 -7.35774630111763E-08 5.31188101282609E-08"
        rpy="0 0 0" />
      <mass
        value="2.39970051632588" />
      <inertia
        ixx="0.00501929933650507"
        ixy="-2.25989622798604E-08"
        ixz="4.81371943453209E-08"
        iyy="0.0157117164030385"
        iyz="2.29259649253453E-09"
        izz="0.0126583434717946" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_1_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.067 0.106 0.643 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_1_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin1_1"
    type="revolute">
    <origin
      xyz="0.179528059776071 0 0"
      rpy="-1.5707963267949 0 0" />
    <parent
      link="finger_1" />
    <child
      link="finger_1_1" />
    <axis
      xyz="-5.98131790295024E-05 0 -0.999999998211192" />
    <limit
      lower="0"
      upper="2"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_1_1_1">
    <inertial>
      <origin
        xyz="0.0632003612386414 3.99101890424802E-08 -4.97915925157244E-09"
        rpy="0 0 0" />
      <mass
        value="1.21959091463891" />
      <inertia
        ixx="0.0027989422341273"
        ixy="1.18563005691453E-09"
        ixz="1.03187407648895E-09"
        iyy="0.00236364981492399"
        iyz="9.47605149827563E-12"
        izz="0.00235597916354427" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_1_1_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.929 0.953 0.251 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_1_1_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin1_1_1"
    type="revolute">
    <origin
      xyz="0.244804461442329 0 0"
      rpy="0 0 0" />
    <parent
      link="finger_1_1" />
    <child
      link="finger_1_1_1" />
    <axis
      xyz="5.98131790298099E-05 0 0.999999998211192" />
    <limit
      lower="-2"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_1_eef">
    <inertial>
      <origin
        xyz="0.0225328824570412 -4.97915925157244E-09 -4.00492903285254E-08"
        rpy="0 0 0" />
      <mass
        value="1.21959091463891" />
      <inertia
        ixx="0.0027989422341273"
        ixy="1.03187407649125E-09"
        ixz="-1.18563005723979E-09"
        iyy="0.00235597916354427"
        iyz="-9.47605149824953E-12"
        izz="0.00236364981492399" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_1_eef.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.929 0.953 0.251 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_1_eef.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin1_eef"
    type="fixed">
    <origin
      xyz="0.0406674787816002 0 0"
      rpy="1.5707963267949 0 0" />
    <parent
      link="finger_1_1_1" />
    <child
      link="finger_1_eef" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="finger_2">
    <inertial>
      <origin
        xyz="0.0582113997401504 7.2059313627193E-08 -1.65269150809166E-07"
        rpy="0 0 0" />
      <mass
        value="2.62291216939469" />
      <inertia
        ixx="0.0059916892852438"
        ixy="2.54236843542237E-08"
        ixz="-6.34236507666107E-08"
        iyy="0.0153587785697811"
        iyz="-1.78838763283136E-07"
        izz="0.0188064904153421" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.067 0.106 0.643 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin2"
    type="revolute">
    <origin
      xyz="0.365640976139111 0 0.74425746223111"
      rpy="3.14159265358979 0.436413643527903 0" />
    <parent
      link="base_link" />
    <child
      link="finger_2" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1"
      upper="1"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_2_1">
    <inertial>
      <origin
        xyz="0.109229816295533 4.52412628471066E-08 2.04530046343798E-07"
        rpy="0 0 0" />
      <mass
        value="2.39969352647452" />
      <inertia
        ixx="0.00501928036081044"
        ixy="-3.5054689130295E-08"
        ixz="-2.33595141983811E-09"
        iyy="0.0157115485437255"
        iyz="1.64460168230235E-09"
        izz="0.012658197491362" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_2_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.067 0.106 0.643 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_2_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin2_1"
    type="revolute">
    <origin
      xyz="0.179528059776071 0 0"
      rpy="-1.5707963267949 0 0" />
    <parent
      link="finger_2" />
    <child
      link="finger_2_1" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-2"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_2_1_1">
    <inertial>
      <origin
        xyz="0.0632003108555788 9.00354358845945E-08 -5.92834292767463E-09"
        rpy="0 0 0" />
      <mass
        value="1.21958916107902" />
      <inertia
        ixx="0.00279893854396813"
        ixy="2.53112184423615E-09"
        ixz="1.00694975972186E-09"
        iyy="0.00236365044723376"
        iyz="1.55441919145354E-11"
        izz="0.00235597934140297" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_2_1_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.929 0.953 0.251 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_2_1_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin2_1_1"
    type="revolute">
    <origin
      xyz="0.244804461442329 0 0"
      rpy="0 0 0" />
    <parent
      link="finger_2_1" />
    <child
      link="finger_2_1_1" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="2"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_2_eef">
    <inertial>
      <origin
        xyz="0.0225328320739786 -5.92834292767463E-09 -9.01745371706397E-08"
        rpy="0 0 0" />
      <mass
        value="1.21958916107902" />
      <inertia
        ixx="0.00279893854396813"
        ixy="1.00694975979725E-09"
        ixz="-2.53112184418194E-09"
        iyy="0.00235597934140297"
        iyz="-1.55441919151448E-11"
        izz="0.00236365044723376" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_2_eef.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.929 0.953 0.251 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_2_eef.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin2_eef"
    type="fixed">
    <origin
      xyz="0.0406674787816002 0 0"
      rpy="1.5707963267949 0 0" />
    <parent
      link="finger_2_1_1" />
    <child
      link="finger_2_eef" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="finger_3">
    <inertial>
      <origin
        xyz="0.0582113837489124 4.16223523425074E-08 4.90502328753806E-08"
        rpy="0 0 0" />
      <mass
        value="2.62291219261925" />
      <inertia
        ixx="0.00599150094431563"
        ixy="2.04568885804215E-08"
        ixz="1.56227476638942E-08"
        iyy="0.0153590426966865"
        iyz="-1.69856752749064E-07"
        izz="0.0188065552886401" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.067 0.106 0.643 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin3"
    type="revolute">
    <origin
      xyz="0.330644040002145 -0.175 1.18915890545841"
      rpy="3.14159265358979 -0.58248657962397 0" />
    <parent
      link="base_link" />
    <child
      link="finger_3" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1"
      upper="1"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_3_1">
    <inertial>
      <origin
        xyz="0.109230136263842 -9.75973253325435E-08 1.96832685372961E-07"
        rpy="0 0 0" />
      <mass
        value="2.39970180172285" />
      <inertia
        ixx="0.00501928275652698"
        ixy="-1.3940007970864E-08"
        ixz="6.66458995034566E-11"
        iyy="0.0157116553919968"
        iyz="4.10856664670477E-10"
        izz="0.0126583003760249" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_3_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.067 0.106 0.643 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_3_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin3_1"
    type="revolute">
    <origin
      xyz="0.179528059776071 0 0"
      rpy="-1.5707963267949 0 0" />
    <parent
      link="finger_3" />
    <child
      link="finger_3_1" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="0"
      upper="2"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_3_1_1">
    <inertial>
      <origin
        xyz="0.0632003618512376 4.0027509862206E-08 -4.9791595291282E-09"
        rpy="0 0 0" />
      <mass
        value="1.21959090651552" />
      <inertia
        ixx="0.00279894205576526"
        ixy="1.24368845267914E-09"
        ixz="1.76726335114814E-09"
        iyy="0.00236364600043913"
        iyz="2.18626444917931E-11"
        izz="0.0023559755663535" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_3_1_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.929 0.953 0.251 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_3_1_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin3_1_1"
    type="revolute">
    <origin
      xyz="0.244804461442329 0 0"
      rpy="0 0 0" />
    <parent
      link="finger_3_1" />
    <child
      link="finger_3_1_1" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="0"
      upper="2"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="finger_3_eef">
    <inertial>
      <origin
        xyz="0.0225328830696374 -4.9791595291282E-09 -4.01666112592736E-08"
        rpy="0 0 0" />
      <mass
        value="1.21959090651552" />
      <inertia
        ixx="0.00279894205576526"
        ixy="1.76726335122045E-09"
        ixz="-1.24368845289598E-09"
        iyy="0.0023559755663535"
        iyz="-2.18626444909685E-11"
        izz="0.00236364600043913" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_3_eef.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.929 0.953 0.251 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://dexterous/meshes2/finger_3_eef.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="fin3_eef"
    type="fixed">
    <origin
      xyz="0.0406674787816004 0 0"
      rpy="1.5707963267949 0 0" />
    <parent
      link="finger_3_1_1" />
    <child
      link="finger_3_eef" />
    <axis
      xyz="0 0 0" />
  </joint>
</robot>